application/src/main/java/org/togetherjava/tjbot/features/help/HelpSystemHelper.java

```diff
@@ -176,17 +178,29 @@ RestAction<Message> constructChatGptAttempt(ThreadChannel threadChannel,
             return useChatGptFallbackMessage(threadChannel);
         }
 
+        List<String> ids = new CopyOnWriteArrayList<>();
         RestAction<Message> message =
                 mentionGuildSlashCommand(threadChannel.getGuild(), ChatGptCommand.COMMAND_NAME)
                     .map("""
                             Here is an AI assisted attempt to answer your question 🤖. Maybe it helps! \
                             In any case, a human is on the way 👍. To continue talking to the AI, you can use \
                             %s.
                             """::formatted)
-                    .flatMap(threadChannel::sendMessage);
-
-        for (String aiResponse : chatGPTAnswer.get()) {
-            message = message.map(aiResponse::formatted).flatMap(threadChannel::sendMessage);
+                    .flatMap(threadChannel::sendMessage)
+                    .onSuccess(m -> ids.add(m.getId()));
+
+        String[] answers = chatGPTAnswer.get();
+        for (int i = 0; i < answers.length; i++) {
+            if (i == answers.length - 1) {
+                message = message.map(answers[i]::formatted)
+                    .flatMap(m -> threadChannel.sendMessage(m)
+                        .addActionRow(Button.danger(componentIdInteractor
+                            .generateComponentId(ids.toArray(String[]::new)), "Dismiss")));
+                continue;
+            }
+            message = message.map(answers[i]::formatted)
+                .flatMap(threadChannel::sendMessage)
+                .onSuccess(m -> ids.add(m.getId()));
```

Comment:
2 remarks:
* i dont see `ids.add(..)` on the last message, intentional?
* i dont like the code duplication of the "map formatted, send" part. and it kinda trips me when reading the code for the first time, since i wouldnt know what that block is about (adding the dismiss button)

u can get rid of the code duplication by refactoring it sth like this:
```java
message = message.map(answers[i]::formatted)
  .flatMap(threadChannel.sendMessage);

if (i == answers.length - 1) {
  message = message.addActionRow(...);
}

message = message.onSuccess(...);
```

Reply:
ah ok yeah that makes more sense

Reply:
and yeah the missing onSuccess on the last message is intentional, I get it through the Button event later

Reply:
fair. with my changes, u would likely get it in the list as well. but shouldnt hurt :)